scripts/apogee/apogee.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import datasets
+from datasets import Features, Value, Sequence
+from datasets.data_files import DataFilesPatternsDict
+import itertools
+import h5py
+import numpy as np
+
+# TODO: Add BibTeX citation
+# Find for instance the citation on arxiv or on the dataset repo/website
+_CITATION = """\
+@InProceedings{huggingface:dataset,
+title = {A great new dataset},
+author={huggingface, Inc.
+},
+year={2020}
+}
+"""
+
+# TODO: Add description of the dataset here
+# You can copy an official description
+_DESCRIPTION = """\
+Stellar spectra dataset based on SDSS-IV APOGEE.
+"""
+
+# TODO: Add a link to an official homepage for the dataset here
+_HOMEPAGE = ""
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = ""
+
+_VERSION = "0.0.1"
+
+# Full list of features available here:
+# https://data.sdss.org/datamodel/files/APOGEE_ASPCAP/APRED_VERS/ASPCAP_VERS/allStar.html
+_FLOAT_FEATURES = [
+    "teff",
+    "logg",
+    "m_h",
+    "alpha_m",
+    "radial_velocity",
+    "restframe",
+]
+
+# Features that correspond to ugriz fluxes
+_FLUX_FEATURES = []
+_BOOL_FEATURES = []
+
+
+class SDSS(datasets.GeneratorBasedBuilder):
+    """TODO: Short description of my dataset."""
+
+    VERSION = _VERSION
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="apogee",
+            version=VERSION,
+            data_files=DataFilesPatternsDict.from_patterns(
+                {"train": ["../apogee/healpix=*/*.hdf5"]}
+            ),
+            description="SDSS APOGEE survey spectra.",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "apogee"
+
+    @classmethod
+    def _info(self):
+        """Defines the features available in this dataset."""
+        # Starting with all features common to spectral dataset
+        features = {
+            "spectrum": Sequence(
+                {
+                    "flux": Value(dtype="float32"),
+                    "ivar": Value(dtype="float32"),
+                    # "lsf_sigma": Value(dtype="float32"),
```

Comment:
is it that the lsf is constant? and/or the lsf is not very important here?

For uniformity, we can still have an `lsf_sigma` entry, but set it to a constant value. The main reason for doing that is that this way, a neural network built to support lsf information will also understand this type of data.

Reply:
I have added a rough estimate on the lsf. lsf is not important here.